src/backend/renderer/element/mod.rs

```diff
@@ -100,6 +101,21 @@ pub enum UnderlyingStorage {
     /// A wayland buffer
     #[cfg(feature = "wayland_frontend")]
     Wayland(Buffer),
+    /// A memory backed buffer
+    Memory {
+        /// Data of the underlying buffer
+        data: *const u8,
```

Comment:
So I don't think I have seen any null-ptr checks for this anywhere.

I assume there is a problem with using a proper lifetime here? If so, that's fine, but can we make this at least a little bit more safe?

Reply:
It is only used to get a reference to the underlying storage of an element. If the element can't provide
some underlying storage it should return `None` and never a null-ptr. I can add some checks for that, but
I don't they are necessary and may also just be some debug_asserts.

Yeah, we can not (easily) introduce a proper lifetime handling here. What we could do imo, is to introduce
some small wrapper around some memory representing pixel data that can be cloned around. It could
be made COW and used in the memory element and returned in the underlying storage. That would be
way safer as it should guarantee that the memory is accessible for as long as you hold a reference to it.

Reply:
>  It could
be made COW and used in the memory element and returned in the underlying storage. That would be
way safer as it should guarantee that the memory is accessible for as long as you hold a reference to it.


That sounds like a better solution. I am fine, if you prefer to do that in a follow up PR though.

Reply:
okay, I drafted that and pushed a commit showing off the `MemoryBuffer`. 
If this is an acceptable solution I would fixup the other commits with these changes.

Reply:
Looks like a good approach to me! :+1:

Reply:
done